vmm/src/device_manager.rs

```diff
@@ -2751,6 +2744,11 @@ impl DeviceManager {
                     .map_err(DeviceManagerError::CreateVirtioMem)?,
                 ));
 
+                // Update the virtio-mem zone so that it has a handle onto the
+                // virtio-mem device, which will be used for triggering a resize
+                // if needed.
+                virtio_mem_zone.set_virtio_device(Arc::clone(&virtio_mem_device));
```

Comment:
Not 100% sure about the architectural cleanliness of the coupling between `MemoryManager` and now `virtio_devices::Mem`

Reply:
It's not much different than before given the Resize structure was defined in `virtio_devices/src/mem.rs`. And it makes sense IMO because `virtio-mem` is very much tied to the memory management. It's a bit of a special device that can be considered more than a regular device and therefore it's okay not to be entirely managed by the device manager.